src/python/WMCore/REST/Server.py

```diff
@@ -767,9 +767,19 @@ def _call(self, param):
             raise APINotSpecified()
         api = param.args.pop(0)
         if api not in self.methods[request.method]:
-            response.headers['Allow'] = \
-                " ".join(sorted([m for m, d in self.methods.iteritems() if api in d]))
-            raise APIMethodMismatch()
+            methods = " ".join(sorted([m for m, d in self.methods.iteritems() if api in d]))
+            response.headers['Allow'] = methods
+            foundMethod = None
```

Comment:
```suggestion
            if not methods:
                msg = 'Api "%s" not found. This method supports these Apis: %s' % (api, self.methods[request.method])
                raise APINotSupported(msg)
            else:
                msg = 'Api "%s" only supported in the method: "%s"' % (api, methods)
                raise APIMethodMismatch(msg)
```

how about the code above?

Reply:
well, and deleting a few lines below (first time I make this suggestion instead of a comment).

Reply:
done